cmd/agent/common/autoconfig.go

```diff
@@ -32,14 +32,11 @@ func SetupAutoConfig(confdPath string) {
 	// NOTICE: this will also setup the Python environment, if available
 	Coll = collector.NewCollector(GetPythonPaths()...)
 
-	// create the Autoconfig instance
-	AC = autodiscovery.NewAutoConfig(Coll)
+	// registering the check scheduler
+	scheduler.Register("check", collector.InitCheckScheduler(Coll))
```

Comment:
any reason to register the schedulers on the `scheduler` package instead of registering them on a  `MetaScheduler` struct that'd be passed on to the AC constructor? It'd sound cleaner to me, but might be missing something

Reply:
A lot better agreed, and as the `MetaScheduler` implement the `Scheduler` interface we can just pass a `Scheduler` to the AC 👍